nemo/collections/nlp/data/intent_slot_classification/intent_slot_classification_dataset.py

```diff
@@ -59,6 +59,11 @@ def get_features(
 
         for j, word in enumerate(words):
             word_tokens = tokenizer.text_to_tokens(word)
+
+            # to handle emojis that could be neglected during tokenization
+            if len(word.strip()) > 0 and len(word_tokens) == 0:
+                word_tokens = [tokenizer.ids_to_tokens(tokenizer.unk_id)]
```

Comment:
Does this work for say consecutive multiple emojis (without space or something to seperate as "word")

Reply:
to extract each emoji we would need to use a package like [ this ](https://pypi.org/project/emoji/)
I do not think it's general enough to add in the processing given the fact that emojies are missing from HF LM vocabs.

Reply:
@carolmanderson any opinion on that?

Reply:
Ok so for now, if we dont use that package, a string of consecutive emojis would flatten out to a single unk token. As long as that doesnt hurt the model during training or inference, I am ok with that

Reply:
I think it's good as written. If we have a "word" consisting of multiple emojis, and it fails to generate any subtokens,  I think it's fine to replace that whole word with UNK. If users want to split up runs of multiple emojis, they can do that as a preprocessing step.

Reply:
Sounds good. Approved